include/drivers/gpio.h

```diff
@@ -356,6 +356,25 @@ struct gpio_dt_spec {
 		.pin = DT_GPIO_PIN(node_id, prop),			\
 		.dt_flags = DT_GPIO_FLAGS(node_id, prop),		\
 	}
+/**
+ * @brief Like GPIO_DT_SPEC_GET(), with a fallback to a default value
+ *
+ * If the devicetree node identifier 'node_id' refers to a node with a
+ * property 'prop', this expands to <tt>GPIO_DT_SPEC_GET(node_id, prop)</tt>.
+ * The @p default_value parameter is not expanded in this case.
+ *
+ * Otherwise, this expands to @p default_value.
+ *
+ * @param node_id devicetree node identifier
+ * @param prop lowercase-and-underscores property name
+ * @param default_value fallback value to expand to
+ * @return static initializer for a struct gpio_dt_spec for the property,
+ *         or default_value if the node or property do not exist
+ */
+#define GPIO_DT_SPEC_GET_OR(node_id, prop, default_value) \
```

Comment:
I'm not sure if this macro will be ever used for cases other than

```c
GPIO_DT_SPEC_GET_OR(node_id, prop, {0})
```

So the macro could be designed without the need of explicitly defining it.

Reply:
> So the macro could be designed without the need of explicitly adding `{0}`.

Please review the discussion in  #31280; I'm doing it the way the user asked for it.

Reply:
It's really a matter of taste, but I'd provide another macro that avoids passing `{0}` (which could be based on the current one).

Reply:
I'd rather not do it that way; I think the number of combinations will get too large when your PR #34539 is merged. This is very readable IMO and it is consistent with all the other DT-ish `THING_OR(node, prop, default)` APIs that explicitly take a `default` value.